collector-examine

Dennok opened the pull request:
Now collector on examine shows real produced power and stored power.
Rename last_power to stored_power.

:cl:
add: Collectors show their real produced power and stored power when examined
/:cl:

[why]: Now it show only stored power.
Now : 
![screenshot_1](https://user-images.githubusercontent.com/7734424/40491953-19ca65b8-5f78-11e8-9ef5-157a5bc70663.png)
![screenshot_2](https://user-images.githubusercontent.com/7734424/40491956-1bfe43d6-5f78-11e8-9b5c-f70094732b32.png)

MrDoomBringer commented:
>bubblegum slams into you
i see you follow standard NT collector testing procedures as well

no but really good change

Tlaltecuhtli commented:
its probably an hallucination

Tlaltecuhtli commented:
you should test if it works while soloing a bubblegun tho

PKPenguin321 commented:
Delete "optional name here" from :cl:

SpaceManiac reviewed (changes requested):
One grammar item, otherwise looks good.

Dennok commented on code/modules/power/singularity/collector.dm:
```diff
@@ -174,9 +174,9 @@
 	if(active)
 		if(!bitcoinmining)
-			to_chat(user, "<span class='notice'>[src]'s display states that it is processing <b>[DisplayPower(last_power)]</b>.</span>")
+			to_chat(user, "<span class='notice'>[src]'s display states that it is stored <b>[DisplayPower(stored_power)]</b>, and processing <b>[DisplayPower(min(stored_power, (stored_power*RAD_COLLECTOR_STORED_OUT)+1000))]</b>.</span>")
```
Ready

Dennok replied:
@SpaceManiac

Cyberboss commented on code/modules/power/singularity/collector.dm:
```diff
@@ -174,9 +174,9 @@
+			to_chat(user, "<span class='notice'>[src]'s display states that it has stored <b>[DisplayPower(stored_power)]</b>, and processing <b>[DisplayPower(min(stored_power, (stored_power*RAD_COLLECTOR_STORED_OUT)+1000))]</b>.</span>")
 		else
-			to_chat(user, "<span class='notice'>[src]'s display states that it is producing a total of <b>[last_power*RAD_COLLECTOR_MINING_CONVERSION_RATE]</b> research points per minute.</span>")
+			to_chat(user, "<span class='notice'>[src]'s display states that it has stored a total of <b>[stored_power*RAD_COLLECTOR_MINING_CONVERSION_RATE]</b>, and producing [(min(stored_power, (stored_power*RAD_COLLECTOR_STORED_OUT)+1000))*RAD_COLLECTOR_MINING_CONVERSION_RATE] research points per minute.</span>")
```
Maybe create a function for this equation since it seems to be used in a few places

Dennok commented on code/modules/power/singularity/collector.dm:
```diff
@@ -65,9 +66,9 @@
-			last_power-=bitcoins_mined
+			var/bitcoins_mined = RAD_COLLECTOR_OUTPUT*RAD_COLLECTOR_MINING_CONVERSION_RATE
+			SSresearch.science_tech.add_point_type(TECHWEB_POINT_TYPE_DEFAULT, bitcoins_mined)
+			stored_power-=bitcoins_mined
```
Oh, yes.You right.

Dennok replied:
Fixed

Dennok committed:
collector-examine

Now collector on examine shows real produced power and stored power.
Rename last_power to stored_power.
